homeconnect/api.py

```diff
@@ -195,13 +195,13 @@ def handle_event(self, event, appliance):
         if event.event == "NOTIFY" or event.event == "STATUS":
             appliance.status.update(data_dict)
 
-        if event.event == "CONNECTED":
+        elif event.event == "CONNECTED":
             appliance.connected = True
-        if event.event == "DISCONNECTED":
+        elif event.event == "DISCONNECTED":
             appliance.connected = False
 
         if appliance.event_callback is not None:
-            appliance.event_callback(appliance)
+            appliance.event_callback(appliance, event.event, data_dict)
```

Comment:
Isn't changing the signature of the callback a breaking change that could break downstream code?

Reply:
It is a breaking change, but the code implementing the event_callback should change accordingly.

Reply:
But isn't it unfriendly for a library to do this? At least it would require a major version bump.

What about accepting a tuple as callback argument and dispatching depending on whether an appliance or tuple is received?

Reply:
How would that work?

Reply:
Yeah good question, sorry that didn't make sense.

What aboout just

```python
try:
    appliance.event_callback(appliance, event.event, data_dict)
except TypeError:
    appliance.event_callback(appliance)
```
?

Reply:
I changed it! Thanks!